lib/plugins/threaded-signing/plugin.c

```diff
@@ -163,10 +163,11 @@ signing_worker_thread(void *user_data)
       assert(self->signature_info->hash);
       memcpy(self->signature_info->hash, self->input_buffer[0], self->hash_size);
 
-      free(self->input_buffer[0]);
       self->input_buffer[0]= NULL;
```

Comment:
This will leak memory, since you do not free. You should reuse the memory for later by moving the pointer to the first available slot in input_buffer.